models/project/project.go

```diff
@@ -61,6 +62,89 @@ func FindProject(projectID int64) *ProjectStruct {
 	}
 }
 
+func Create(userId int64, title string, description string, repositoryID int64, repositoryOwner string, repositoryName string, oauthToken sql.NullString) (p *ProjectStruct, e error) {
+	objectDB := &db.Database{}
+	var interfaceDB db.DB = objectDB
+	table := interfaceDB.Init()
+	defer table.Close()
+	tx, _ := table.Begin()
+	defer func() {
+		if err := recover(); err != nil {
+			logging.SharedInstance().MethodInfo("Project", "Create").Error("unexpected error")
+			tx.Rollback()
+			e = errors.New("unexpected error")
+			p = nil
+		}
+	}()
+
+	var repoId sql.NullInt64
+	var repo *repository.RepositoryStruct
+	if repositoryID != 0 {
+		repo = repository.NewRepository(0, repositoryID, repositoryOwner, repositoryName)
+		if !repo.Save() {
```

Comment:
__[gometalinter]__
* [gotype] invalid operation: repo (variable of type *invalid type) has no field or method Save